app/src/androidTest/java/com/github/geohunt/app/GoogleMapsActivityTest.kt

```diff
@@ -0,0 +1,60 @@
+package com.github.geohunt.app
+
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import androidx.test.rule.GrantPermissionRule
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiObject
+import androidx.test.uiautomator.UiSelector
+import org.hamcrest.CoreMatchers.notNullValue
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class GoogleMapsActivityTest {
+    @get:Rule
+    val composeTestRule = createAndroidComposeRule<GoogleMapsActivity>()
+
+    @get:Rule
+    val permissionRuleFineLocation: GrantPermissionRule = GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION)
+
+    @get:Rule
+    val permissionRuleCoarseLocation: GrantPermissionRule = GrantPermissionRule.grant(android.Manifest.permission.ACCESS_COARSE_LOCATION)
+
+    @Test
+    fun testMapDisplayed() {
+        onView(withId(R.id.map_container_view)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun testMapIsNotNull() {
+        onView(withId(R.id.map_container_view)).check(matches(notNullValue()))
+    }
+
+    @Test
+    fun testMarkerClickOnEvent1WorksCorrectly() {
+        val device: UiDevice = UiDevice.getInstance(getInstrumentation())
+        val marker: UiObject = device.findObject(UiSelector().descriptionContains("Event 1"))
+        marker.click()
+    }
+
+    @Test
+    fun testMarkerClickOnEvent2WorksCorrectly() {
+        val device: UiDevice = UiDevice.getInstance(getInstrumentation())
+        val marker: UiObject = device.findObject(UiSelector().descriptionContains("Event 2"))
+        marker.click()
+    }
+
+    @Test
+    fun testMarkerClickOnEvent3WorksCorrectly() {
+        val device: UiDevice = UiDevice.getInstance(getInstrumentation())
+        val marker: UiObject = device.findObject(UiSelector().descriptionContains("Event 3"))
+        marker.click()
+    }
```

Comment:
Theses methods do not test what happens after the click event. There is no check for the correct display of the marker information.

Reply:
I don't think that testing that the application does not crash on click is functionnally the same as testing the correct display of the marker's information, as you could totally have no interaction on the click, and still pass the test. Ideally, we should check that it is displayed in some form, but I do understand that testing maps manually is a pain and that it is very hard to test for the content of the canvas.

Sorry if this was unrealistic, I thought it would be easier to test 😅